code/game/objects/items/stacks/stack.dm

```diff
@@ -172,13 +172,31 @@
 
 /obj/item/stack/proc/use(var/amount)
 	ASSERT(isnum(src.amount))
+
 	if(src.amount>=amount)
 		src.amount-=amount
 	else
 		return 0
 	. = 1
-	if (src.amount<=0)
+	if (src.amount<=0) //If the stack is empty after removing the required amount of items!
 		if(usr)
+			if(istype(usr,/mob/living/silicon/robot))
+				var/mob/living/silicon/robot/R=usr
+				if(R.module)
+					R.module.modules -= src
+				if(R.module_active == src) R.module_active = null
+				if(R.module_state_1 == src)
```

Comment:
yes it's sad but there are actually 3 different variables for the item slots
I'd try to change that but it will take some time (and this PR wants to be merged asap, since engiborgs are horribly bugged and potentially unplayable right now)